autopush/waker.py

```diff
@@ -0,0 +1,101 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from urllib import urlencode
+import requests
+import time
+
+""" Client disconnect and remote wake. """
+
+
+class WakeException(Exception):
+    pass
+
+
+class IWake(object):
+    """ Wake interface
+
+    Several protocol extensions provide for an external "wake" option.
+    For instance, a vendor provided UDP wake extension for simplePush allows
+    servers to use a proprietary system that sends a UDP based wake message
+    to devices. This message bears no data or version information and only
+    wakes a device that may not be actively connected to the network.
+
+    """
+
+    def __init__(self, *args, **kwargs):
+        """ Setup """
+
+    def register(self, wake_info=None):
+        """ Endpoint Registration """
+
+    def set_active(self):
+        """Set period to time out and the destruction method"""
+
+    def check_idle(self):
+        """ Check if the connection has idled out """
+
+    def send_wake(self, wake_info=None):
+        """ Send a wake notification to the device """
+
+
+class UDPWake(IWake):
+    """ Proprietary UDP wake tool """
+
+    data = {}
+    default_period = 10 * 1000  # default to 10 seconds
+    endpoint = None     # Third party wake endpoint
+    timeout = 0
+    idle = 0
+
+    def _ms_time(self):
```

Comment:
We should prolly stick this in one spot and use it from there. My bad, I'll open a ticket for that.